### DISCUSSION TITLE
Trigger resize

### DISCUSSION THREAD

**Discussion by lordcraymen** [google/model-viewer#4435] [Q&A] [answered]
Is it possible to provoke a resize with JavaScript?

I managed to transpile and polyfill  three and model-viewer to work on iOS 11 iPhone 6s.

I guess the model viewer does not adjust to parents with changing height because iOS 11 does not have an IntersectionObserver yet. Although I polyfilled it, obviously it’s not reactive and I’d have to set it to polling, which I’d like to avoid. Is there a way to „tickle“ model viewer to make it resize? I could call a JavaScript after the parent ob the viewer has resized.

**Comment by elalish** [comment, marked as answer]
It's actually based on a `ResizeObserver`, and in fact we already have a built-in fallback if that's not present. I would search our code for:

```
    if (HAS_RESIZE_OBSERVER) {
      this[$resizeObserver]!.observe(this);
    } else {
      self.addEventListener('resize', this[$fallbackResizeHandler]);
    }
```
to debug what's going wrong.

**Reply by lordcraymen** [reply]
Thanks, I’ll look into that.
